crates/prune/prune/src/segments/user/storage_history.rs

```diff
@@ -65,7 +62,7 @@ where
         };
         if limiter.is_limit_reached() {
             return Ok(SegmentOutput::not_done(
-                PruneInterruptReason::new(&limiter),
+                PruneLimiter::interrupt_reason(&limiter),
```

Comment:
let's replace these with `limiter.interrupt_reason()`

Reply:
Ah yes, it's much better.